### PR TITLE
Add info about configuring and running third-party tracers.

### PR DESCRIPTION
**Pull request by stbof**
https://cockroachlabs.atlassian.net/browse/DOC-214

### CODE REVIEW

**Comment by stbof** [comment]
> nit: please squash the multiple commits into one (`git rebase -i` -> squash) unless you actually want them as separate commits.

In the doc team we keep them as separate commits and squash on merge.

**Review by andreimatei — APPROVED** [review]
:LGTM:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13336#-:-MzpFhhUCackhjoiQL2P:bmxilld)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei and @stbof)

___
*[v22.1/show-trace.md, line 207 at r3](https://reviewable.io/reviews/cockroachdb/docs/13336#-MzpA1jH2w_ZbDEmZsHN:-MzpA1jH2w_ZbDEmZsHO:bykh29j) ([raw file](https://github.com/cockroachdb/docs/blob/066412f5672aba3abb965911a1fd52de5c960725/v22.1/show-trace.md#L207)):*
> ```Markdown
> ## Route traces to a third-party collector
> 
> You can configure CockroachDB to send traces to a third-party collector. CockroachDB supports [Jaeger](https://www.jaegertracing.io/), [Lightstep](https://lightstep.com/), and [Zipkin](https://zipkin.io/). Enabling tracing also activates all the log messages, at all verbosity levels, as traces include the log messages printed in the respective trace context.
> ```

Instead of mentioning Lightstep explicitly, I think we should say Jaeger, Zipkin, and any trace collector that  can ingest traces over the standard OTLP protocol.
Jaeger and Zipkin are special because we can output to their native formats; we don't have any special Lightstep support (we used to, and we also used to have special DataDog support, but we removed them when we added the otlp stuff).
___
*[v22.1/show-trace.md, line 224 at r3](https://reviewable.io/reviews/cockroachdb/docs/13336#-MzpCvhQ0DjVGqj865Wm:-MzpCvhQ0DjVGqj865Wn:bstka8z) ([raw file](https://github.com/cockroachdb/docs/blob/066412f5672aba3abb965911a1fd52de5c960725/v22.1/show-trace.md#L224)):*
> ```Markdown
> <thead><tr><th>Setting</th><th>Type</th><th>Default</th><th>Description</th></tr></thead>
> <tbody>
> <tr><td><code>trace.debug.enable</code></td><td>boolean</td><td><code>false</code></td><td>If set, traces for recent requests can be seen at <code>https://&ltui&gt;/debug/requests</code>.</td></tr>
> ```

Instead of talking of `/debug/requests`, I would talk about `/#/debug/tracez`. This page is new in 22.1. As opposed to `/debug/requests`, `tracez` has more functionality and it's where the development will be from now on. I will delete the `debug/requests` code at some point.
`tracez` doesn't have historical information (it only has information on spans that are currently open). But it's much cheaper than `trace.debug.enabled`, and it's enabled by default (through `trace.span_registry.enabled`). At the moment, the page only lists the spans currently open on the node rendering the page, but that will hopefully change soon to show information from across the cluster.

File: v22.1/show-trace.md [hunk 1]
```diff
@@ -38,21 +33,20 @@ For `SHOW TRACE FOR SESSION`, no privileges are required.
...
 
-To further clarify these concepts, let's look at a visualization of a trace for one statement. This particular trace is [visualized by Jaeger](query-behavior-troubleshooting.html#visualize-statement-traces-in-jaeger). The image shows spans and log messages. You can see names of operations and sub-operations, along with parent-child relationships and timing information, and it's easy to see which operations are executed in parallel.
+Consider a visualization of a trace for one statement as [visualized by Jaeger](query-behavior-troubleshooting.html#visualize-statement-traces-in-jaeger). The image shows spans and log messages. You can see names of operations and sub-operations, along with parent-child relationships and timing information, and it's easy to see which operations are executed in parallel.
```
Comment (rmloveland) [inline v22.1/show-trace.md:49]: Appreciate this concision

File: v22.1/show-trace.md [hunk 2]
```diff
@@ -197,19 +190,76 @@ In this example, we use two terminals concurrently to generate conflicting trans
...
       ...
-    (200 rows)
     ~~~
 
+## Route traces to a third-party collector
+
+You can configure CockroachDB to send traces to a third-party collector. CockroachDB supports [Jaeger](https://www.jaegertracing.io/), [Zipkin](https://zipkin.io/), and any trace collector that can ingest traces over the standard OTLP protocol. Enabling tracing also activates all the log messages, at all verbosity levels, as traces include the log messages printed in the respective trace context.
+
+{{site.data.alerts.callout_info}}
+Enabling full tracing is expensive both in terms of CPU usage and memory footprint, and is not suitable for high throughput  production environments.
```
Comment (rmloveland) [inline v22.1/show-trace.md:210]: There appear to be two spaces between 'throughput' and 'production' but maybe it's just the web UI I'm looking at

File: v22.1/show-trace.md [hunk 3]
```diff
@@ -197,19 +190,76 @@ In this example, we use two terminals concurrently to generate conflicting trans
...
+{{site.data.alerts.end}}
+
+You can configure the CockroachDB tracer to route to the OpenTelemetry tracer, with OpenTelemetry being supported by all observability tools. In particular, you can configure CockroachDB to output traces to:
+
+- A collector, such as Lightstep and special builds of Jaeger, that uses the OpenTelemetry Protocol (OTLP).
```
Comment (rmloveland) [inline v22.1/show-trace.md:215]: is Lightstep something we can link to?  sorry if you already did somewhere and I missed it

File: v22.1/show-trace.md [hunk 4]
```diff
@@ -197,19 +190,76 @@ In this example, we use two terminals concurrently to generate conflicting trans
...
+    ~~~ sql
+    SET CLUSTER SETTING trace.jaeger.agent='localhost:6831'
+    ~~~
+
+1. Go to [http://localhost:16686](http://localhost:16686).
+1. Select the CockroachDB service, and view traces streaming in.
+
+Instead of searching through log messages in an unstructured fashion, the logs are now graphed in a tree format based on how the contexts were passed around. This also traverses machine boundaries so you don't have to look at different flat `.log` files to correlate events.
+
+Jaeger's memory storage works well for small use cases, but can result in out of memory errors when collecting many traces over a long period of time. Jaeger also supports disk-backed local storage using Badger. To use this, start Jaeger by running the following Docker command:
```
Comment (rmloveland) [inline v22.1/show-trace.md:254]: Is Badger a thing we can link to?  from this context it seems like some kind of BerkeleyDB-type storage thing, yes?

**Review by kevin-v-ngo — APPROVED** [review]
Looks good to me! Thanks Stephanie.

**Review by rmloveland — APPROVED** [review]
LGTM

**Pull request merged**